runtime-modules/content-directory/src/lib.rs

```diff
@@ -1478,18 +1490,27 @@ decl_module! {
             //
 
             // Decrease reference counters of involved entities (if some)
-            if let Some(entity_ids_to_decrease_rcs) = property_value_vector.get_vec_value().get_involved_entities() {
+            let entities_inbound_rcs_delta = if let Some(entity_ids_to_decrease_rcs) =
```

Comment:
This set of steps is quite involved, could you perhaps 
a) move it out into a standalone function, and just have that function compute the final `entities_inbound_rcs_delta`, so that reading the main extrisnics body is a bit easier, and the reader can go and read only this function if they want to make sure its correct.
c) in that function, can you break things up a bit. For example you are doing a two function calls, a `Option` match, and an `if` body computation, all in one line (1493). This can be broken up in steps for greater readability. It also is a bit rough with different variabls that have the same name, but different type, in different near by scopes.

Reply:
a) Done here https://github.com/Joystream/joystream/pull/715/commits/4272f54e4a93a23e0bf001a1744ecbbf09863a04
c) https://github.com/Joystream/joystream/pull/715/commits/103f0e61e0523918801db2d9f7253ed981d3db73